servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/RetryingHttpRequesterFilter.java

```diff
@@ -232,10 +232,12 @@ protected Single<StreamingHttpResponse> request(final StreamingHttpRequester del
             if (responseMapper != null) {
                 single = single.flatMap(resp -> {
                     final HttpResponseException exception = responseMapper.apply(resp);
-                    return exception != null ?
+                    return (exception != null ?
                             // Drain response payload body before discarding it:
-                            resp.payloadBody().ignoreElements().onErrorComplete().concat(Single.failed(exception)) :
-                            Single.succeeded(resp);
+                            resp.payloadBody().ignoreElements().onErrorComplete()
+                                    .concat(Single.<StreamingHttpResponse>failed(exception)) :
+                            Single.succeeded(resp))
+                            .shareContextOnSubscribe();
```

Comment:
would it make a difference if the shareContextOnSubscribe is inside our side of the flatMap?

Reply:
I'm not sure I understood the question, feel free to continue if I misunderstood.
Because the function returns an async source, `flatMap` will subscribe to it. `shareContextOnSubscribe()` should be the last operator in the chain so that subscribe happens directly to `shareContextOnSubscribe()` to have any effect.